model/src/main/scala/ch.epfl.scala.index.model/SemanticVersion.scala

```diff
@@ -68,20 +68,40 @@ case class SemanticVersion(
 }
 
 object SemanticVersion extends Parsers {
-  private implicit val preReleaseOrdering: Ordering[PreRelease] = Ordering.by {
-    case ReleaseCandidate(rc) => (Some(rc), None, None)
-    case Milestone(m) => (None, Some(m), None)
-    case OtherPreRelease(pr) => (None, None, Some(pr))
-  }
+  private implicit val preReleaseOrdering: Ordering[Option[PreRelease]] =
+    Ordering.by {
+      case None                       => (3, None, None)
+      case Some(ReleaseCandidate(rc)) => (2, Some(rc), None)
+      case Some(Milestone(m))         => (1, Some(m), None)
+      case Some(OtherPreRelease(pr))  => (0, None, Some(pr))
+    }
 
-  implicit val ordering: Ordering[SemanticVersion] = Ordering.by {
-    x => (x.major, x.minor, x.patch, x.patch2, x.preRelease)
+  implicit def ordering: Ordering[SemanticVersion] = Ordering.by { x =>
```

Comment:
Why use a `def` here? That doesn't seem like an improvement to me.

Reply:
Indeed